readthedocs/notifications/signals.py

```diff
@@ -0,0 +1,45 @@
+"""Custom Django signals related to notifications."""
+
+import structlog
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from readthedocs.notifications.models import Notification
+from readthedocs.organizations.models import Organization
+from readthedocs.projects.models import Project
+from readthedocs.projects.notifications import MESSAGE_PROJECT_SKIP_BUILDS
+from readthedocs.subscriptions.notifications import MESSAGE_ORGANIZATION_DISABLED
+
+log = structlog.get_logger(__name__)
+
+
+@receiver(post_save, sender=Project)
+def project_skip_builds(instance, *args, **kwargs):
+    """Check if the project is ``skip`` and add/cancel the notification."""
+    if instance.skip:
+        Notification.objects.add(
+            message_id=MESSAGE_PROJECT_SKIP_BUILDS,
+            attached_to=instance,
+            dismissable=False,
+        )
+    else:
+        Notification.objects.cancel(
```

Comment:
Seems this we're going to be calling this query a lot on projects that were never skipped. We don't save projects super frequently, but it seems less than ideal to add overhead like this for very rare outcomes. 

Can we cancel the notification on project same if the `skip` field has been changed or similar?

Reply:
> Can we cancel the notification on project same if the `skip` field has been changed or similar?

I'm not sure how we can detect this. `pre_save` and `post_save` signals receive a `update_fields` (https://docs.djangoproject.com/en/5.0/ref/signals/#post-save) but it's only passed when using `.save(update_fields=[...])`, which we don't use.

I remember we were checking if some field was updated in a form and perform something in particular, but I'm not finding it. However, that approach won't be useful here either because this won't work when the field is updated _outside_ that particular form (Django Admin or Django Shell).

Do you an idea in mind here?

Reply:
I think this approach is fine for now, we just need to monitor performance. It's likely not a real issue.